### PR TITLE
Chainer 3.0.0

### PR DESCRIPTION
**Pull request by Hakuyume**
- [x] Update README
- [x] Update travis config
- [x] `Variable.data` -> `Variable.array` (https://github.com/chainer/chainer/pull/3223)
- [x] Use `as_variable` (https://github.com/chainer/chainer/pull/3218)
- [x] Use `ConcatenatedDataset` (https://github.com/chainer/chainer/pull/2780)
- [x] Use `cupy.argsort` (https://github.com/cupy/cupy/pull/67, https://github.com/cupy/cupy/pull/288)
- [x] Remove `if` statement related to reduction bug (https://github.com/cupy/cupy/pull/505)
- [x] Use `apply()` instead of `__call__()` for `MaxPooling2D` (https://github.com/chainer/chainer/pull/3257)
- [x] Install `PyTest` for `testing.attr` (https://github.com/chainer/chainer/pull/3324)

### CODE REVIEW

**Comment by yuyu2172** [comment]
SegNet test fails due to Chainer 3.0.0 change.

**Comment by Hakuyume** [comment]
@yuyu2172 Thank you. I fixed it. Currently, Travis CI does not run any tests of `SegNet` because they are marked as `slow`.

**Comment by Hakuyume** [comment]
@yuyu2172

File: README.md [hunk 1]
```diff
@@ -43,8 +43,9 @@ For additional features
 
 Environments under Python 2.7.12 and 3.6.0 are tested.
 
-+ The master branch will work on both the stable version (v2) and the development version (v3).
++ The master branch will work on both the stable version (v3) and the development version (v4).
 + For users using Chainer v1, please use version `0.4.11`, which can be installed by `pip install chainercv==0.4.11`. This branch is unmaintained.
```
Comment (yuyu2172) [inline README.md]: How about not repeating the sentence "This branch is unmaintained".

**Comment by yuyu2172** [comment]
Sorry for being late!

**Comment by yuyu2172** [comment]
Should we add attributes for all methods or just one for each class?
I prefer declaring attr.slow for a class if all tests are slow because there are less code.

**Comment by Hakuyume** [comment]
> Should we add attributes for all methods or just one for each class?
> I prefer declaring attr.slow for a class if all tests are slow because there are less code.

I also prefer declaring `attr.slow` for a class. However, it does not work since Chainer v3.1. I guess this is related to migration to PyTest.

**Comment by yuyu2172** [comment]
I see.
This would force us to change other tests too.

**Comment by Hakuyume** [comment]
Yes. I changed other tests, too.

**Comment by yuyu2172** [comment]
LGTM

**Pull request merged**